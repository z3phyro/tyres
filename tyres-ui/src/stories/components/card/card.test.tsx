import { test, expect } from "vitest";
import { render } from "@solidjs/testing-library";
import Card from "./card";

test("Card renders", () => {
  const { getByRole } = render(() => <Card />);

  const card = getByRole("article") as HTMLElement;
  expect(card).toBeTruthy();
});

test("Card renders with custom content", () => {
  const { getByRole } = render(() => <Card>Card content</Card>);

  const card = getByRole("article");
  expect(card.textContent).toBe("Card content");
});

test("Card renders with custom class", () => {
  const { getByRole } = render(() => <Card class="my-custom-class">Card content</Card>);

  const card = getByRole("article");
  expect(card).toHaveClass("my-custom-class");
});

test("Card snapshot test", () => {
  const { asFragment } = render(() => <Card>Card content</Card>);
  expect(asFragment()).toMatchSnapshot();
});
