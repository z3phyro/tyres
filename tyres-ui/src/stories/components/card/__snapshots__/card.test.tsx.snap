// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card snapshot test 1`] = `"<article class="shadow rounded-lg bg-white p-6 mb-4 undefined">Card content</article>"`;
